ci/d.yml

```
@@ -17,7 +17,7 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
-    - uses: dlang-community/setup-dlang@7c3e57bdc1ff2d8994f00e61b3ef400e67d2d7ac
+    - uses: dlang-community/setup-dlang@v1
```

Could we keep the SHA hash and leave a comment letting users opt-in to pinning to a version/branch?
Similar to the [Ruby workflow](https://github.com/actions/starter-workflows/blob/main/ci/ruby.yml)

```suggestion
    # To automatically get bug fixes and new versions for dlang-community/setup-dlang,
    # change this to:
    # uses: dlang-community/setup-dlang@v1
    - uses: dlang-community/setup-dlang@4c99aa991ce7d19dd3064de0a4f2f6b2f152e2d7
```

---

> Keeping on upgrading the hash will provide a terrible user experience,
> as it is likely to be forgotten, and will also require user-intervention
> when another set of change breaks this action, hence why this updates
> the hash to a version instead of changing it to the latest hash.

As explained in the commit message, I think this provides a terrible user-experience. Most users, if not all, will want the auto-updating behavior, which the Github team even acknowledge with their guidelines of have `v1` branch.
Having a comment to tell user to do the thing they want just relegates community-maintained languages, or languages Github has little interest in, to a back-seat, as obvious from the time this PR (and its predecessor) has been open.